### PR TITLE
Bump Lein versions and replace images

### PR DESCRIPTION
**Pull request by dragonsahead**
Bumped Lein 2.8 with 2.9.5
Change the build image from CircleCI to OpenJDK:alpine to build and test

### CODE REVIEW

**Comment by paoliniluis** [comment]
I don't know if this will work @camsaul, how can I test a build without any caching?

File: .circleci/config.yml [hunk 1]
```diff
@@ -66,24 +60,24 @@ executors:
...
-          MB_DB_USER: metabase_test
-          MB_POSTGRESQL_TEST_USER: metabase_test
-      - image: postgres:latest
+          MB_DB_DBNAME: circle_test
+          MB_DB_USER: circle_test
+          MB_POSTGRESQL_TEST_USER: circle_test
+      - image: circleci/postgres:latest
         environment:
-          POSTGRES_USER: metabase_test
-          POSTGRES_DB: metabase_test
```
Comment (camsaul) [inline .circleci/config.yml:80]: does this make any difference?

**Review by camsaul — APPROVED** [review]
This looks ok to me. If you put `[ci nocache]` in the commit message it will run everything with no cached results -- you should do that before merging this

**Comment by paoliniluis** [comment]
@camsaul I will skip the ci as the tests that are not passing are due to the snowflake issue running out of resources

**Comment by paoliniluis** [comment]
Well, I can't skip it seems (which is good), but I can't merge. So I'll leave it up to you

**Comment by paoliniluis** [comment]
@ariya just made a new tag to pin it on the builds

**Pull request merged**